framework/graphics/fps_info.cpp

```diff
@@ -90,19 +102,71 @@ void FpsInfo::BeginFrame(uint64_t frame)
         {
             measurement_start_time_ = util::datetime::GetTimestamp();
             started_measurement_    = true;
+            frame_durations_.clear();
         }
     }
+
+    frame_start_time_ = util::datetime::GetTimestamp();
 }
 
 void FpsInfo::EndFrame(uint64_t frame)
 {
     if (started_measurement_ && !ended_measurement_)
     {
+        frame_durations_.push_back(util::datetime::DiffTimestamps(frame_start_time_, util::datetime::GetTimestamp()));
+
         // Measurement frame range end is non-inclusive, as opposed to trim frame range
         if (frame >= measurement_end_frame_ - 1)
         {
             measurement_end_time_ = util::datetime::GetTimestamp();
             ended_measurement_    = true;
+
+            // Save measurements to file
+            if (!measurement_file_name_.empty())
+            {
+                double   start_time   = util::datetime::ConvertTimestampToSeconds(measurement_start_time_);
+                double   end_time     = util::datetime::ConvertTimestampToSeconds(measurement_end_time_);
+                double   diff_time    = GetElapsedSeconds(measurement_start_time_, measurement_end_time_);
+                uint64_t total_frames = measurement_end_frame_ - measurement_start_frame_;
+                double   fps          = static_cast<double>(total_frames) / diff_time;
+
+                nlohmann::json file_content = { { "frame_range",
+                                                  { { "start_frame", measurement_start_frame_ },
+                                                    { "end_frame", measurement_end_frame_ },
+                                                    { "frame_count", total_frames },
+                                                    { "start_time_monotonic", start_time },
+                                                    { "end_time_monotonic", end_time },
+                                                    { "duration", diff_time },
+                                                    { "fps", fps },
+                                                    { "frame_durations", frame_durations_ } } } };
+
+                FILE*   file_pointer = nullptr;
+                int32_t result       = util::platform::FileOpen(&file_pointer, measurement_file_name_.c_str(), "w");
+                if (result == 0)
+                {
+                    const std::string json_string = file_content.dump(4);
```

Comment:
We have a `kJsonIndentWidth` if you want to be consistent with json output elsewhere. Up to you.

Reply:
Done, but I just had to make kJsonIndentWidth visible for that (Put it as a static constexpr in the header)